src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -34,162 +44,541 @@ public abstract class IdentifiersShouldBeSpelledCorrectlyAnalyzer : DiagnosticAn
+            customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+            FileParseRule,
+            AssemblyRule,
+            NamespaceRule,
+            TypeRule,
+            VariableRule,
+            MemberRule,
+            MemberParameterRule,
+            DelegateParameterRule,
+            TypeTypeParameterRule,
+            MethodTypeParameterRule,
+            AssemblyMoreMeaningfulNameRule,
+            NamespaceMoreMeaningfulNameRule,
+            TypeMoreMeaningfulNameRule,
+            MemberMoreMeaningfulNameRule,
+            MemberParameterMoreMeaningfulNameRule,
+            DelegateParameterMoreMeaningfulNameRule,
+            TypeTypeParameterMoreMeaningfulNameRule,
+            MethodTypeParameterMoreMeaningfulNameRule);
+
         public override void Initialize(AnalysisContext analysisContext)
-#pragma warning restore RS1025 // Configure generated code analysis
         {
             analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        private static CodeAnalysisDictionary GetMainDictionary()
+        {
+            var assemblyType = typeof(IdentifiersShouldBeSpelledCorrectlyAnalyzer);
+            var assembly = assemblyType.GetTypeInfo().Assembly;
+            var dictionary = $"{assemblyType.Namespace}.Dictionary.dic";
+
+            using var stream = assembly.GetManifestResourceStream(dictionary);
+            var text = SourceText.From(stream);
+            return ParseDicDictionary(text);
+        }
+
+        private static CodeAnalysisDictionary ParseXmlDictionary(SourceText text)
+            => text.Parse(CodeAnalysisDictionary.CreateFromXml);
+
+        private static CodeAnalysisDictionary ParseDicDictionary(SourceText text)
+            => text.Parse(CodeAnalysisDictionary.CreateFromDic);
+
+        private static string RemovePrefixIfPresent(string prefix, string name)
+            => name.StartsWith(prefix, StringComparison.Ordinal) ? name.Substring(1) : name;
+
+        private static IEnumerable<Diagnostic> GetUnmeaningfulIdentifierDiagnostics(ISymbol symbol, string symbolName)
+        {
+            switch (symbol.Kind)
+            {
+                case SymbolKind.Assembly:
+                    yield return Diagnostic.Create(AssemblyMoreMeaningfulNameRule, Location.None, symbolName);
+                    break;
+
+                case SymbolKind.Namespace:
+                    yield return Diagnostic.Create(NamespaceMoreMeaningfulNameRule, symbol.Locations.First(), symbolName);
+                    break;
+
+                case SymbolKind.NamedType:
+                    foreach (var location in symbol.Locations)
```

Comment:
This will end up reporting duplicate diagnostics per each partial declaration. Instead we just report diagnostic for symbol's first location. You should be able to change the method to return `Diagnostic` instead of `IEnumerable<Diagnostic>`

Reply:
Report first location now